frontend/src/components/modals/settings/SettingsForm.tsx

```diff
@@ -46,11 +53,46 @@ function SettingsForm({
         ariaLabel="model"
         items={models.map((model) => ({ value: model, label: model }))}
         defaultKey={settings.LLM_MODEL || models[0]}
-        onChange={onModelChange}
+        onChange={(e) => {
+          const key = localStorage.getItem(
+            `API_KEY_${settings.LLM_MODEL || models[0]}`,
+          );
+          onAPIKeyChange(key || "");
+          onModelChange(e);
+        }}
         tooltip={t(I18nKey.SETTINGS$MODEL_TOOLTIP)}
         allowCustomValue // user can type in a custom LLM model that is not in the list
         disabled={disabled}
       />
+      <Input
+        label="API Key"
+        disabled={disabled}
+        aria-label="apikey"
+        data-testid="apikey"
+        placeholder={t(I18nKey.SETTINGS$API_KEY_PLACEHOLDER)}
+        type={isVisible ? "text" : "password"}
+        value={apiKey}
+        onChange={(e) => {
```

Comment:
Maybe you can outsource this to a handler defined in the parent. This component is more for displaying data rather than mixing and handling business logic.

Something like this:

```ts
const handleAPIKeyChange = (key: string) => {
  localStorage.setItem(`API_KEY_${settings.LLM_MODEL || models[0]}`, key);
  setApiKey(key);
};
```

(and if needed we can extend the function to handle set/get cases)